tee-worker/litentry/core/parachain-extrinsic-task/receiver/src/lib.rs

```diff
@@ -0,0 +1,56 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+extern crate sgx_tstd as std;
+
+use itp_extrinsics_factory::CreateExtrinsics;
+use itp_ocall_api::EnclaveOnChainOCallApi;
+use itp_types::parentchain::ParentchainId;
+use lc_parachain_extrinsic_task_sender::init_parachain_extrinsic_sender_storage;
+use log::*;
+use std::{format, string::String, sync::Arc, time, vec};
+
+const MAX_BATCH_SIZE: usize = 500;
+const BATCH_EXTRINSIC_INTERVAL: time::Duration = time::Duration::from_secs(6);
+
+pub fn run_parachain_extrinsic_task_receiver<ExtrinsicsFactory, OCallApi>(
+	api: Arc<OCallApi>,
+	extrinsic_factory: Arc<ExtrinsicsFactory>,
+) -> Result<(), String>
+where
+	ExtrinsicsFactory: CreateExtrinsics + Send + Sync + 'static,
+	OCallApi: EnclaveOnChainOCallApi,
+{
+	let task_receiver = init_parachain_extrinsic_sender_storage().map_err(|e| {
+		format!("Failed to initialize parachain extrinsic task sender storage: {:?}", e)
+	})?;
+	let mut calls = vec::Vec::new();
+
+	loop {
+		let start_time = time::Instant::now();
+		while start_time.elapsed() < BATCH_EXTRINSIC_INTERVAL && calls.len() < MAX_BATCH_SIZE {
+			if let Ok(call) = task_receiver.recv() {
+				calls.push(call);
+			}
+		}
+		if !calls.is_empty() {
```

Comment:
So there is a possible scenario here, when you exit the above loop, that is the `recv()`, the vc task might still be sending extrinsics unaware that the receiver has exit the loop and this thread is in the process of submitting extrinsic, this will result in `Disconnected` error when the vc task tries submitting the extrinsic. Maybe a unit test to ensure that might not happen. 

Can the while condition be convert to an `if` and the `if` condition be converted to `while`, So something like this: 
```rust
while let Ok(call) = task_receiver.recv() { 
      calls.push(call)
      if start_time.elapsed() > BATCH_EXTRINSIC_INTERVAL && .. { 
            // submit extrinsics here 
      }
}
```

Reply:
I thought about this, This is not a good idea too, We may be waiting for too long since this if condition is only triggered when it receives the message. So please ignore, i think the current logic should be fine